components/pipedream/sql/sql.js

```diff
@@ -0,0 +1,82 @@
+const pipedream = require("https://github.com/PipedreamHQ/pipedream/components/pipedream/pipedream.app.js");
+
+module.exports = {
+  name: "pipedream-sql",
+  version: "0.0.1",
+  dedupe: "unique", // Dedupes records based on the query execution ID
+  props: {
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        cron: "0 0 * * *",
+      },
+    },
+    sqlQuery: {
+      type: "string",
+      label: "SQL Query",
+      description:
+        "Your SQL query (try running in [https://pipedream.com/sql](https://pipedream.com/sql) first)",
+    },
+    resultType: {
+      type: "string",
+      label: "Result Type",
+      description: `Specifies how you want the query results formatted`,
+      optional: true,
+      options: ["array", "object", "csv"],
+      default: "array",
+    },
+    emitEachRecordAsEvent: {
+      type: "boolean",
+      label: "Emit each record as its own event",
+      description:
+        "If `true`, each record in your results set is emitted as its own event. Defaults to emitting results as a single event (an array of records)",
+      optional: true,
+      default: false,
+    },
+    pipedream,
+  },
+  async run() {
+    const results = await this.pd.runSQLQuery(this.sqlQuery, this.resultType);
+    console.log(results);
```

Comment:
Should we be concerned about result size? Both in terms of emits and also log observations? Does `runSQLQuery` limit? Looking at that code now.

Reply:
Talked with G about this and noted the limitations in the README. Thinking about a better way to enforce on the rt / Athena side, too.